From 2e4125c85c87cc01e621db0730c9896703ec5ad7 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Fri, 26 Jul 2013 14:13:12 +0800
Subject: [PATCH 12/12] ENGR00272743-1 Add param_set_max() as it been removed
 in jb4.3

Add param_set_max() as it will be used in pcm_get_near_param()

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 pcm.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/pcm.c b/pcm.c
index f5eafe5..541ff86 100644
--- a/pcm.c
+++ b/pcm.c
@@ -113,6 +113,14 @@ static unsigned int param_get_max(struct snd_pcm_hw_params *p, int n)
     return 0;
 }
 
+static void param_set_max(struct snd_pcm_hw_params *p, int n, unsigned int val)
+{
+    if (param_is_interval(n)) {
+        struct snd_interval *i = param_to_interval(p, n);
+        i->max = val;
+    }
+}
+
 static void param_set_int(struct snd_pcm_hw_params *p, int n, unsigned int val)
 {
     if (param_is_interval(n)) {
-- 
1.8.0

